routers/web/repo/repo.go

```diff
@@ -576,23 +577,61 @@ func SearchRepo(ctx *context.Context) {
 		return
 	}
 
-	results := make([]*api.Repository, len(repos))
+	// collect the latest commit of each repo
+	repoIDsWithCommitStatuses := make([]int64, 0, len(repos))
+	commitsWithCommitStatuses := make([]string, 0, len(repos))
+	for _, repo := range repos {
+		branches, branchCount, err := repo_service.GetBranches(ctx, repo, 0, 0)
```

Comment:
It would be better if there was method to get default branch directly not get all branches and than loop

Reply:
Done